src/resources/views/crud/columns/inc/column_wrapper.blade.php

```diff
@@ -0,0 +1,28 @@
+@php
```

Comment:
Excellent. Great job @pxpm . Can we use this for the missing column types too?

Reply:
@pxpm I'm wondering - wouldn't it be easier to add wrapper support to columns if, instead of loading this ```column_wrapper``` blade file inside each column, we:
- move all this PHP logic inside ```wrapperStart```, 
- inside the column we do, for example for the ```phone``` column:

```php
@php
  // -----------------
  // phone column type
  // -----------------
  $column['value'] = data_get($entry, $column['name']);
  $column['text'] = Str::limit(strip_tags($column['value']), array_key_exists('limit', $column) ? $column['limit'] : 40, "[...]");
  $column['escaped'] = $column['escaped'] ?? true;
  if ($column['escaped') {
      $column['text'] = htmlspecialchars($column['text']);
  }

  // for columns that need a default wrapper
  $column['wrapper'] = $column['wrapper'] ?? [
      'element' => 'a',
      'href' => 'mailto:'.$column['text'],
  ]
@endphp

<span>
    @include('crud::columns.inc.wrapperStart')
    {!! $column['text'] !!}
    @include('crud::columns.inc.wrapperEnd')
</span>
```

I think what makes it difficult to add this functionality to columns is the fact that the "text" has to be nice and clean, in order to fit ```column_wrapper```. But if we split the wrapper in two parts - ```wrapperStart``` and ```wrapperEnd```, we can put that start and end before _anything_. Even crazy stuff like a ```table``` and such.

Not sure I'm explaining this correctly. Let me know.

Reply:
Hello @tabacitu 

I agree with you, providing defaults for the columns that has them atm. I like your solution to those columns.

But I don't agree with removing the `@include('column_wrapper')`, by your example we need to add the escaped check to every column, (lot's of repetition added) and do some crazy stuff like escaping with `htmlspecialchars` and them show the unescaped result. It just seems ... mehh!

But i might not be seeing the whole picture. From my understanding column_wrapper will output whatever you send there, just check for escape or not escape. 

Let me know if I am mistaken.

Reply:
In terms of escaping using ```htmlspecialchars()``` ourselves, I don't think it's a bad thing. As far as I know, that's exactly what Laravel does behind the scenes different between ```{{ }}``` and ```{!! !!}```. So whether we do ```{{ $something }}``` or ```{!! htmlspecialchars($something) }}```, the resulting PHP code will be exactly the same. No performace loss. No parsing difference. No difference in output.

---

I do agree it's a bit... "meh" to have the escape calculation inside each field type, though... 😆 But the way I see it:

CONs:
- The logic would be repeated inside quite a few columns.

PROs:
- To avoid the repeated logic, we _could_ do something like below (though I'm not sure it's necessary for something this simple that gets repeated):

```php
// crud/columns/inc/escapeColumnText.blade.php
  if ($column['escaped') {
      $column['text'] = htmlspecialchars($column['text']);
  }
```
```php
@php
  // -----------------
  // phone column type
  // -----------------
  $column['value'] = data_get($entry, $column['name']);
  $column['text'] = Str::limit(strip_tags($column['value']), array_key_exists('limit', $column) ? $column['limit'] : 40, "[...]");
  $column['escaped'] = $column['escaped'] ?? true;

  @include('crud::columns.inc.escapedColumnText')

  // for columns that need a default wrapper
  $column['wrapper'] = $column['wrapper'] ?? [
      'element' => 'a',
      'href' => 'mailto:'.$column['text'],
  ]
@endphp

<span>
    @include('crud::columns.inc.wrapperStart')
    {!! $column['text'] !!}
    @include('crud::columns.inc.wrapperEnd')
</span>
```
- Currently in the PR, the ```escaped``` logic is inside the ```wrapper``` logic; so you can't use ```escaped``` for any column that doesn't also support ```wrapper```; which is a bit "meh" in itself; why would the two functionalities get mixed together if we can keep them apart?
- Not ALL of the columns make sense to have the ```escaped``` functionality. So the ```escaped``` logic is currently performed even for those that don't really make sense;
- But the biggest PRO of making this change, and I guess the reason why this jumped at me, is that by doing the actual outputting inside a secondary view we include, we're obscuring something very important about how the column works; the most important thing, actually; so if you have no idea how Columns work in Backpack, when you open up a column, if you see this at the end:
```php
<span>
    @include('crud::columns.inc.wrapperStart')
    {!! $column['text'] !!}
    @include('crud::columns.inc.wrapperEnd')
</span>
```
I think it would be much clearer to understand how it works than if you see:
```php
<span>
         @include('crud::columns.inc.column_wrapper')
 </span>
```
Because it's right there - you see what the actual output of the column is. It's obvious that it's just wrapped inside and element. So when you think about creating a new column, you instantly understand that you can just change ```$column['text']``` above and that's how you end up with a different output. I think I should have started with this issue 🤣 but until now it was just a bad feeling, an itch. Now that I put it into words does it make more sense to do start&end - what do you think?

Reply:
Actually now that I look at it, the above could easily become:

```php
@php
  // -----------------
  // phone column type
  // -----------------
  $column['value'] = data_get($entry, $column['name']);

  // trim the output to 40 characters or the given limit
  $column['text'] = Str::limit(strip_tags($column['value']), array_key_exists('limit', $column) ? $column['limit'] : 40, "[...]");

  // escape the output by default, unless otherwise defined
  $column['text'] = ($column['escaped'] ?? true) ? htmlspecialchars($column['text']) : $column['text'];

  // for columns that need a default wrapper
  $column['wrapper'] = $column['wrapper'] ?? [
      'element' => 'a',
      'href' => 'mailto:'.$column['text'],
  ]
@endphp

<span>
    @include('crud::columns.inc.wrapperStart')
    {!! $column['text'] !!}
    @include('crud::columns.inc.wrapperEnd')
</span>
```

So just like the ```limit``` functionality that some columns have, some don't. 
And the ```prefix``` and ```suffix``` functionality that some colums have, and some don't. 

This ```escaped``` functionality, consisting of one line, would be repeated in some columns, but not in others. 

I there we're ok with that...

Reply:
>But the biggest PRO of making this change, and I guess the reason why this jumped at me, is that by doing the actual outputting inside a secondary view we include, we're obscuring something very important about how the column works; the most important thing, actually; 

Yeah, this is the point. Now I got it. 

I will be on this right after I finish what I am working on now. 

Thanks for the input and clarification @tabacitu :+1: